src/services/publisher_service/helpers.rs

```diff
@@ -154,12 +155,14 @@ pub async fn pick_subscriber_notification_for_processing(
     Ok(notification)
 }
 
-#[instrument(skip(postgres))]
+#[instrument(skip(postgres, metrics))]
 pub async fn update_message_processing_status(
     notification: Uuid,
     status: SubscriberNotificationStatus,
     postgres: &PgPool,
+    metrics: Option<&Metrics>,
 ) -> std::result::Result<(), sqlx::error::Error> {
+    let ctx = Context::current();
```

Comment:
non-blocking:

Not sure exactly what a `Context` does so maybe there is a reason this should be here, but in the rest of our notify-server code the `Context::current()` call comes just before invoking the metrics themselves. [E.g.](https://github.com/WalletConnect/notify-server/blob/08a6b6d5338721381fb771de60bfa4cb8e2767bb/src/metrics.rs#L205)

Perhaps this could be moved inside `update_metrics_on_message_status_change()` to avoid an extra function param?

Reply:
As we are using the context just to satisfy metrics call it makes sense just move it into the function. Done.